app/src/components/modal/modal_deposit_withdraw/index.tsx

```diff
@@ -103,224 +101,200 @@ interface Props extends HTMLAttributes<HTMLDivElement> {
   const isDepositWithdrawDisabled =
-    displayFundAmount.isZero() ||
     !wallet ||
     displayFundAmount.gt(wallet) ||
     displayFundAmount.isZero() ||
-    displayFundAmount.lt(currencySelected === ExchangeCurrency.Dai ? minDaiExchange : minOmenExchange) ||
-    (currencySelected === ExchangeCurrency.Omen &&
-      exchangeType === ExchangeType.deposit &&
-      displayFundAmount.gt(omenAllowance)) ||
-    (currencySelected === ExchangeCurrency.Omen && exchangeType === ExchangeType.withdraw && xDaiBalance?.isZero())
+    displayFundAmount.lt(currencySelected === 'dai' ? minDaiBridgeExchange : minOmniBridgeExchange) ||
+    (exchangeType === ExchangeType.deposit &&
+      currentTokenMainnet &&
+      currentTokenMainnet.allowance &&
+      displayFundAmount.gt(new BigNumber(currentTokenMainnet.allowance))) ||
+    (exchangeType === ExchangeType.withdraw && xDaiBalance?.isZero())
 
   const depositWithdraw = async () => {
     if (!cpk) {
       return
     }
 
     try {
-      setMessage(
-        `${exchangeType} ${formatBigNumber(displayFundAmount || new BigNumber(0), DAI.decimals)} ${
-          currencySelected === ExchangeCurrency.Dai ? DAI.symbol : 'OMN'
-        }`,
-      )
+      setMessage(`${exchangeType} ${formatBigNumber(displayFundAmount || new BigNumber(0), decimals)} ${symbol}`)
       setTxState(TransactionStep.waitingConfirmation)
-      setConfirmations(9)
+      setConfirmations(0)
       setIsTransactionModalOpen(true)
 
       const hash =
         exchangeType === ExchangeType.deposit
-          ? await cpk.sendMainnetTokenToBridge(displayFundAmount, currencySelected)
-          : await cpk.sendXdaiChainTokenToBridge(displayFundAmount, currencySelected)
+          ? await cpk.sendMainnetTokenToBridge(displayFundAmount, address, symbol)
+          : await cpk.sendXdaiChainTokenToBridge(
+              displayFundAmount,
+              address,
+              {
+                setTxState,
+                setTxHash,
+              },
+              symbol,
+            )
 
       const provider = exchangeType === ExchangeType.deposit ? context.rawWeb3Context.library : context.library
 
       setTxNetId(provider.network.chainId)
       setTxHash(hash)
 
-      await waitForConfirmations(hash, provider, setConfirmations, setTxState)
+      await waitForConfirmations(hash, provider, setConfirmations, setTxState, 13)
 
-      if (exchangeType === ExchangeType.deposit && currencySelected !== ExchangeCurrency.Dai) {
+      if (exchangeType === ExchangeType.deposit && symbol !== 'DAI') {
         await XdaiService.waitForBridgeMessageStatus(hash, context.library)
       }
-
-      if (exchangeType === ExchangeType.withdraw && currencySelected !== ExchangeCurrency.Dai) {
+      if (exchangeType === ExchangeType.withdraw && symbol !== 'DAI') {
         await XdaiService.waitForClaimSignature(hash, context.library)
       }
-
+      setTxState(TransactionStep.transactionConfirmed)
       await fetchBalances()
```

Comment:
same here, don't think you need `await` since its return its type is `() => void`